apps/condo/domains/miniapp/schema/preprocessors/B2BAppAccess.js

```diff
@@ -0,0 +1,470 @@
+
+
+function fieldAccessWrapperIfNeeded (access, fnWrapper) {
+    // NOTE: you can use the same object in many places! you don't need to wrap it twice
+    if (!fnWrapper.alreadyprocessedbyb2bappaccessplugin) fnWrapper.alreadyprocessedbyb2bappaccessplugin = true
+
+    const type = getType(access)
+    if (type === 'Boolean') {
+        // No need to wrap! You already have access, or you should not have it anyway!
+        return access
+    } else if (type === 'Function') {
+        // NOTE: to prevent multiple wrapping the same function
+        if (access.alreadyprocessedbyb2bappaccessplugin) return access
+        else return fnWrapper
+    } else if (type === 'AsyncFunction') {
+        // NOTE: to prevent multiple wrapping the same function
+        if (access.alreadyprocessedbyb2bappaccessplugin) return access
+        else return fnWrapper
+    } else if (type === 'Object') {
+        const newAccess = {}
+        if (typeof access.read !== 'undefined') newAccess.read = fieldAccessWrapperIfNeeded(access.read, fnWrapper)
+        if (typeof access.create !== 'undefined') newAccess.create = fieldAccessWrapperIfNeeded(access.create, fnWrapper)
+        if (typeof access.update !== 'undefined') newAccess.update = fieldAccessWrapperIfNeeded(access.update, fnWrapper)
+        if (typeof access.delete !== 'undefined') newAccess.delete = fieldAccessWrapperIfNeeded(access.delete, fnWrapper)
+        if (typeof access.auth !== 'undefined') newAccess.auth = fieldAccessWrapperIfNeeded(access.auth, fnWrapper)
+        return newAccess
+    }
+
+    throw new Error(
+        `fieldAccessWrapperIfNeeded(), received ${type}.`,
+    )
+}
+
+/**
+ *
+ * @param pathToOrganizationId {string[]}
+ * @return {string}
+ *
+ * @example
+ * Input: ['a', 'b', 'c']
+ * Output: '{ a { b { c } } }'
+ */
+const generateGqlDataPart = (pathToOrganizationId) => {
+    if (!pathToOrganizationId || !isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) throw new Error('"pathToOrganizationId" should not be empty array')
+    return `{ ${pathToOrganizationId.join(' { ')}${' }'.repeat(pathToOrganizationId.length)}`
+}
+
+const GENERATED_SCHEMAS = new Map()
+
+/**
+ *
+ * @param pathToOrganizationId {string[]}
+ * @return {*|DocumentNode}
+ *
+ * @example
+ * Input: ['ticket', 'organization', 'id']
+ * Output: gql`
+ *         query getAllTickets ($where: TicketWhereInput, $first: Int = 1) {
+ *             objs: allTickets(where: $where, first: $first) { organization { id } }
+ *         }
+ *     `
+ */
+const generateGql = (pathToOrganizationId) => {
+    if (!pathToOrganizationId || !isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) throw new Error('"pathToOrganizationId" should not be empty array')
+    if (pathToOrganizationId.length < 3) throw new Error(`To generate gql "pathToOrganizationId" must contain at least three elements! But was ${pathToOrganizationId}`)
```

Comment:
Where's this restriction comes from?

Reply:
Otherwise you won't be able to generate gql query

[0]: all<name>s - name query
[1]: field parent item
[2]: field from first children
[3]: field from second children
[...]: field from ... children

There is an example for this function, look at it. 

But yes, you are right, this function can be run with two elements as well

I'll fix it